src/evict/evict_page.c

```diff
@@ -82,7 +83,7 @@ __wt_page_release_evict(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * Track how long the call to evict took. If eviction is successful then
 	 * we have one of two pairs of stats to increment.
 	 */
-	ret = __wt_evict(session, ref, false);
+	ret = __wt_evict(session, ref, false, WT_REF_MEM);
```

Comment:
Seeing this makes me think the code should match what is in `__evict_get_ref`: https://github.com/wiredtiger/wiredtiger/pull/4273/files#diff-e21b7d81bc959ced41989e8a0df26d9fR2217

That is it should consider both WT_REF_LIMBO and WT_REF_MEM pages in the `atomc_cas` and restored state.